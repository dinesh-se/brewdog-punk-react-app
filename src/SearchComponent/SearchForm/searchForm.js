class SearchForm extends React.Component {
  render() {
    return <p>This is from searchFrom component.</p>
  }
}

const appInfo = {
  name: "BrewDog's Beer Application",
  subtitle: "Your beer helper"
};

const searchBeer = (e) => {
  e.preventDefault();
  console.log(e.target.elements);
  const stringValue = e.target.elements.search_input.value;
  const searchBy = e.target.elements.search.value;
  console.log(stringValue, searchBy);
  e.target.elements.search_input.value = '';
};

const jsx = (
  <div>
    <h2>{appInfo.name}</h2>
    {appInfo.subtitle && <p>{appInfo.subtitle}</p>}
    <form onSubmit={searchBeer}>
        <input type="text" id="search_input" placeholder="string" />
        <div>
          <div>
            <input type="radio" id="name" name="search" value="brew_name" defaultChecked/>
            <label htmlFor="name">by name</label>
          </div>
          <div>
            <input type="radio" id="brew" name="search" value="brewed_before"/>
            <label htmlFor="brew">by brewed before date</label>
          </div>
        </div>
        <button type="submit" className="search">Search</button>
      </form>
  </div>
);

const appRoot = document.getElementById('app');

ReactDOM.render(jsx, appRoot);