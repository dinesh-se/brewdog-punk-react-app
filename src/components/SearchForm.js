import React from 'react';
import SearchResults from './SearchResults';

export default class SearchForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      beerList: [],
      isLoaded: false,
      error: null
    };
    this.searchBeer = this.searchBeer.bind(this);
  }

  searchBeer(e) {
    e.preventDefault();
    const stringValue = e.target.elements.search_input.value;
    const searchBy = e.target.elements.search.value;
    this.fetchBeer(searchBy, stringValue)
      .then((response) => {
        this.setState(() => ({
          beerList: response,
          isLoaded: true
        }));
        console.log('beer list', this.state);
      }, (error) => {
        this.setState(() => ({
          error,
          isLoaded: true
        }));
      });
  }

  fetchBeer(type, value) {
    const formattedValue = this.spaceToUnderscore(value);
    return fetch(`https://api.punkapi.com/v2/beers?${type}=${formattedValue}`)
      .then(res => res.json());
  }

  spaceToUnderscore(string) {
    return string.replace(/ /g, '_');
  }

  render() {
    return (
      <div>
        <form onSubmit={this.searchBeer}>
          <input type="text" id="search_input" placeholder="string" />
          <div>
            <div>
              <input type="radio" id="name" name="search" value="brew_name" defaultChecked />
              <label htmlFor="name">by name</label>
            </div>
            <div>
              <input type="radio" id="brew" name="search" value="brewed_before" />
              <label htmlFor="brew">by brewed before date</label>
            </div>
          </div>
          <button type="submit" className="search">Search</button>
        </form>
        <SearchResults beerList={this.state.beerList} />
      </div>
    );
  }
}