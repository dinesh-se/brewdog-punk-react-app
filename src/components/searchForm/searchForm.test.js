import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './searchForm';

test('should render search form correctly', () => {
  const wrapper = shallow(<SearchForm />);

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('.search-input')).toHaveLength(1);
  expect(wrapper.find('.radio-button')).toHaveLength(2);
  expect(wrapper.find('.search-button')).toHaveLength(1);
});

// write test cases to checked radio button and placeholder values