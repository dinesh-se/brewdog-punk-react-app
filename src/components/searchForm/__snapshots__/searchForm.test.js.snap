// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render search form correctly 1`] = `
<div
  className="search-container"
>
  <div
    className="search-form"
  >
    <h3>
      Search
    </h3>
    <form
      className="form-search"
      onSubmit={[Function]}
    >
      <input
        className="search-input"
        id="search_input"
        placeholder="eg: lager"
        type="text"
      />
      <div
        className="search-radio-group"
      >
        <span>
          <input
            className="radio-button"
            defaultChecked={true}
            id="name"
            name="search"
            onChange={[Function]}
            type="radio"
            value="beer_name"
          />
          <label
            className="radio-label"
            htmlFor="name"
          >
            by name
          </label>
        </span>
        <span>
          <input
            className="radio-button"
            id="brew"
            name="search"
            onChange={[Function]}
            type="radio"
            value="brewed_before"
          />
          <label
            className="radio-label"
            htmlFor="brew"
          >
            by brewed before date
          </label>
        </span>
      </div>
      <button
        className="button primary-btn search-button"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
  <SearchResults
    beerList={Array []}
  />
</div>
`;
